addon/-private/system/relationships/state/has-many.js

```diff
@@ -180,13 +180,15 @@ export default class ManyRelationship extends Relationship {
   }
 
   setInitialInternalModels(internalModels) {
-    let args = [0, this.canonicalState.length].concat(internalModels);
-    this.canonicalState.splice.apply(this.canonicalState, args);
-    internalModels.forEach(internalModel => {
-      this.canonicalMembers.add(internalModel);
-      this.members.add(internalModel);
-      this.setupInverseRelationship(internalModel);
-    });
+    if (internalModels && internalModels.length) {
```

Comment:
If it's not undefined but has no length, l think we are in a bad state. We should assert against that in debug maybe, but we shouldn't guard, that would just hide a bug